frontend/src/ui/Blocks/Series.tsx

```diff
@@ -125,25 +125,49 @@ const ReadySeriesBlock: React.FC<ReadyProps> = ({
     const finalTitle = title ?? (showTitle ? series.title : undefined);
 
     const events = series.events.filter(event =>
-        !isPastLiveEvent(event.syncedData?.endTime ?? null, event.isLive));
+        !isPastLiveEvent(event.syncedData?.endTime ?? null, event.isLive)
+        && !isUpcomingEvent(event.syncedData?.startTime ?? null));
+
+    const upcomingEvents = series.events.filter(event =>
+        isUpcomingEvent(event.syncedData?.startTime ?? null));
```

Comment:
Similarly to using the same "when event?" time everywhere, the same "upcoming?" decision should be used everywhere. In particular: whether or not the overlay over the thumbnail should say "Upcoming" (or the time) should use the same logic as this. Otherwise there could be events in the "upcoming box" that don't say "upcoming" in the overlay:

![image](https://user-images.githubusercontent.com/7419664/214064898-c841d4b5-0a42-4f29-88bc-25318dc0dff9.png)

Since the "upcoming box" currently says "upcoming live streams", I suppose you should change the logic here to only include events where `isLive == true`.

Reply:
You are correct, but I somehow don't understand when this would ever happen, besides willfully changing dates in the DB.
Another question for this case is what to do with the "upcoming" non-live event instead. Should it be hidden from the block like past live events are, or show up as the first event after live events? My intuition would be to keep it hidden, but I'm honestly not sure and a little confused regarding the use case.

Reply:
Yeah, you are correct that this should likely never happen. However, we get data from Opencast and it promises us very little about what to expect :/ So it could probably happen.

I think I would not hide these events. They are weird and probably somehow broken, but I would still show them. Outside of the box. I.e. they would likely be the first events outside the box. Your last push hides those events, so I would like to change that still. I think you just need to modify the `events.filter` condition in line 128 to make that happen.

Reply:
I decided to change the `isUpcomingEvent` function instead, which initially also included the check for `isLive` anyway. I hope this is alright with you.

Reply:
Even better!